utilities/pulumi_state_splitter/pulumi_state_splitter/cli.py

```diff
@@ -0,0 +1,124 @@
+"""Command line entry point"""
+
+import functools
+import pathlib
+import subprocess
+import sys
+from typing import Sequence
+
+import click
+
+import pulumi_state_splitter.stored_state
+
+
+@click.group()
+@click.option(
+    "-d",
+    "--backend-directory",
+    default=".",
+    type=click.Path(
+        dir_okay=True,
+        exists=True,
+        file_okay=False,
+        path_type=pathlib.Path,
+        resolve_path=True,
+    ),
+)
+@click.pass_context
+def cli(ctx: click.Context, backend_directory: str):
+    """pulumi_yaml_splitter command line interface."""
+    ctx.obj = backend_directory
+
+
+class StackNameOptionType(click.types.StringParamType):
+    """click option type for Pulumi fully qualified stack names."""
+
+    name = "project-name/stack-name"
+
+    def convert(self, value, param, ctx):
+        return pulumi_state_splitter.stored_state.StackName.from_path(value)
+
+
+def _command(f):
+    @cli.command()
+    @click.option(
+        "-a",
+        "--all-stacks",
+        default=False,
+        is_flag=True,
+    )
+    @click.option(
+        "-s",
+        "--stack",
+        multiple=True,
+        type=StackNameOptionType(),
+    )
+    @click.pass_obj
+    @functools.wraps(f)
+    def check_arguments(backend_dir, all_stacks, stack, **kwargs):
+        if all_stacks and stack:
+            raise ValueError("--all-stacks is mutually exclusive with --stack")
```

Comment:
I am not a big fan of magic values.
I wanted the user to be specific about what they want to happen.
I can reconsider if you think that the current approach is a bad idea.

Reply:
Yep, magic value sounds like a bad idea. On the other hand assuming all stack if you don't specify any might be a good default and it will simplify the code a lot (no more tracking of two different options - just find/discover all stacks if none has been provided). I'll leave this up to you.

Reply:
Not sure, but went for it in 2bda01a14de0eb583b20c9ce96dbc41a1fae322c .